From 91e6cb8dacd3779f2908a603e74720bf26f98da7 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:52 -0400
Subject: [block] blk-mq: remove barrier in bt_clear_tag()

Message-id: <1402670238-13916-65-git-send-email-snitzer@redhat.com>
Patchwork-id: 83822
O-Subject: [RHEL7.1 PATCH 064/150] blk-mq: bitmap tag: remove barrier in bt_clear_tag()
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 60f2df8a29df5f2db2c87fd23122a1cebdf2011a
Author: Ming Lei <tom.leiming@gmail.com>
Date:   Sun May 11 01:01:49 2014 +0800

    blk-mq: bitmap tag: remove barrier in bt_clear_tag()

    The barrier isn't necessary because both atomic_dec_and_test()
    and wake_up() implicate one barrier.

    Signed-off-by: Ming Lei <tom.leiming@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq-tag.c b/block/blk-mq-tag.c
index a81b138..5a83d8e 100644
--- a/block/blk-mq-tag.c
+++ b/block/blk-mq-tag.c
@@ -244,7 +244,6 @@ static void bt_clear_tag(struct blk_mq_bitmap_tags *bt, unsigned int tag)
 
  bs = bt_wake_ptr(bt);
  if (bs && atomic_dec_and_test(&bs->wait_cnt)) {
-  smp_mb__after_clear_bit();
   atomic_set(&bs->wait_cnt, bt->wake_cnt);
   bt_index_inc(&bt->wake_index);
   wake_up(&bs->wait);
-- 
1.7.1